cfme/cloud/keypairs.py

```diff
@@ -110,14 +110,13 @@ def is_displayed(self):
     form = View.nested(KeyPairAddForm)
 
 
-class KeyPairCollection(Navigatable):
+class KeyPairCollection(NavigatableMixin):
     """ Collection object for the :py:class: `cfme.cloud.KeyPair`. """
+    def __init__(self, appliance):
+        self.appliance = appliance
 
     def instantiate(self, name, provider, public_key=None, appliance=None):
-        return KeyPair(name, provider,
-                       public_key=public_key or "",
-                       appliance=appliance or self.appliance,
-                       collection=self)
+        return KeyPair(name, provider, self, public_key=public_key or "")
```

Comment:
REQUIRED: please put parameter names in calls. it makes code more readable and allows to avoid mistakes.